android-o/AutofillFramework/AutofillFramework/CustomVirtualView.cs

```diff
@@ -0,0 +1,547 @@
+                            : null;
+                    case (int) AutofillType.Date:
+                        return AutofillValue.ForDate(date);
+                    default:
+                        return null;
+                }
+            }
+        }
+
+        /**
+		 * A partition represents a logical group of items, such as credit card info.
+		 */
+        public class Partition
+        {
+            public string mName;
+            private SparseArray<Line> mLines = new SparseArray<Line>();
+
+            public Partition(String name)
+            {
+                mName = name;
+            }
+
+            /**
+			 * Adds a new line (containining a label and an input field) to the view.
+			 *
+			 * @param idEntryPrefix id prefix used to identify the line - label node will be suffixed
+			 *                      with {@code Label} and editable node with {@code Field}.
+			 * @param autofillType  {@link View#getAutofillType() autofill type} of the field.
+			 * @param label         text used in the label.
+			 * @param text          initial text used in the input field.
+			 * @param sensitive     whether the input is considered sensitive.
+			 * @param autofillHints list of autofill hints.
+			 * @return the new line.
+			 */
+            public Line AddLine(string idEntryPrefix, int autofillType, string label, string text, bool sensitive,
+                params string[] autofillHints)
+            {
+                Preconditions.CheckArgument(autofillType == (int) AutofillType.Text ||
+                                            autofillType == (int) AutofillType.Date,
+                    "Unsupported type: " + autofillType);
+                Line line = new Line(idEntryPrefix, autofillType, label, autofillHints, text, !sensitive);
+                mVirtualViewGroups.Add(line);
+                int id = line.mFieldTextItem.id;
+                mLines.Put(id, line);
+                mVirtualViews.Put(line.mLabelItem.id, line.mLabelItem);
+                mVirtualViews.Put(id, line.mFieldTextItem);
+                mPartitionsByAutofillId.Put(id, this);
+
+                return line;
+            }
+
+            /**
+			 * Resets the value of all items in the partition.
+			 */
+            public void Reset()
+            {
+                for (int i = 0; i < mLines.Size(); i++)
+                {
+                    mLines.ValueAt(i).Reset();
+                }
+            }
+
+            public override String ToString()
+            {
+                return mName;
+            }
+        }
+
+        public class Line
+        {
+            public View view;
+
+            public Item mFieldTextItem;
+
+            // Boundaries of the text field, relative to the CustomView
+            public Rect mBounds = new Rect();
+
+            public Item mLabelItem;
+            private int mAutofillType;
+
+
+            public Line(string idEntryPrefix, int autofillType, string label, string[] hints, string text,
```

Comment:
We should change the type of the autofillType param from `int` to `AutofillType`.